src/inference_profiler.cc

```diff
@@ -622,12 +622,25 @@ InferenceProfiler::Profile(
   is_stable = false;
   meets_threshold = true;
 
-  RETURN_IF_ERROR(dynamic_cast<RequestRateManager*>(manager_.get())
-                      ->PerformWarmup(request_rate, warmup_request_count));
-  RETURN_IF_ERROR(dynamic_cast<RequestRateManager*>(manager_.get())
-                      ->ChangeRequestRate(request_rate, request_count));
-  std::cout << "Request Rate: " << request_rate
-            << " inference requests per seconds" << std::endl;
+  auto custom_manager =
+      dynamic_cast<CustomRequestScheduleManager*>(manager_.get());
+  auto request_rate_manager = dynamic_cast<RequestRateManager*>(manager_.get());
+
+  if (custom_manager) {
```

Comment:
This `if` flips between essentially the same code, just with different object types from a common pointer type using `dynamic_cast`. Given that `PerformWarmup` and `ChangeRequestRate` are being made virtuals already in this PR, this shouldn't be needed at all. The only difference lies in the debug message being emitted, which could also just become a virtual function returning a string, such as `GetManagerName`, to embed in a status message such as this one. This means `dynamic_cast` here is completely redundant with the rest of the code.

Reply:
I think dynamic cast is required. `RequestRate` inherits from `LoadManage`r class. These function are specific to `RequestRate` and classes that inherits from it. If I remove dynamic cast, the type for manager would be LoadManager which would throw an error when I try to call the `PerformWarmup` and `ChangeRequestRate` functions.

Reply:
Right, this means we only need a single `dynamic_cast` to `RequestRateManager`. We might want to ensure it's not `nullptr` for safety, in case the manager object we get at this point is indeed one of the two types, but otherwise the same remark should apply.